### ISSUE TITLE
File with RDFa validates online but not offline

### ISSUE THREAD

**Issue by christianhujer** [validator/validator#432]
When I validate the attached file `start.xhtml` with file upload to [W3C Validator](http://validator.w3.org/), it works.
When I validate the attached file by URL [URL](http://nelkinda.com/blog/) to [W3C Validator](https://validator.w3.org/nu/?doc=http%3A%2F%2Fnelkinda.com%2Fblog%2F), it works.
When I validate the file from the command line using `java -jar vnu.jar htdocs/blog/start.xhtml`, I get the following error messages:
```
"file:/C:/Users/hujerc/git/nelkinda/website/htdocs/blog/start.xhtml":1.1838-1.2027: error: Attribute “content” not allowed on element “span” at this point.
"file:/C:/Users/hujerc/git/nelkinda/website/htdocs/blog/start.xhtml":1.2858-1.2858: error: Attribute “content” not allowed on element “span” at this point.
"file:/C:/Users/hujerc/git/nelkinda/website/htdocs/blog/start.xhtml":1.3526-1.3715: error: Attribute “content” not allowed on element “span” at this point.
"file:/C:/Users/hujerc/git/nelkinda/website/htdocs/blog/start.xhtml":1.4179-1.4179: info warning: Document uses the Unicode Private Use Area(s), which should not be used in publicly exchanged documents. (Charmod C073)
```

Which leads to the following test scenario:
```
  Scenario: Validating a valid XHTML5 file with structured data in RDFa
    Given file 'RDFa_testcase.xhtml'
    When validating that file
    # Expected behavior
    Then the validator should show the file as valid
    # Actual behavior
    Then the validator reports content attributes as invalid
```

**Comment by christianhujer** [comment]
File start.xhtml:
```
<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE html><html xmlns="http://www.w3.org/1999/xhtml" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:og="http://ogp.me/ns#" xml:lang="en"><head><link href="/css/style" rel="stylesheet"/><link href="/images/favicon3_16.png" rel="icon" type="image/png"/><meta name="description" content="Nelkinda Software Craft Blog with articles around Software Craftsmanship"/><title>Blog</title></head><body typeof="Blog" vocab="http://schema.org/"><header><h1 class="title"><a href="/">NELKINDA SOFTWARE CRAFT</a></h1><div class="navigation"><div class="container"><nav class="headernav"><ul class="mainMenu nav nav-pills"><li><a href="/"><i class="fa fa-home"/> Home</a></li><li><a href="/training/"><i class="fa fa-graduation-cap"/> Training</a></li><li><a href="/events/"><i class="fa fa-calendar"/> Events</a></li><li><a href="/career/"><i class="fa fa-user-plus"/> Career</a></li><li><a href="/products/"><i class="fa fa-download"/> Products</a></li><li><a href="/news/"><i class="fa fa-newspaper-o"/> News</a></li><li><a href="/about/"><i class="fa fa-users"/> About Us</a></li><li><a href="/contact"><i class="fa fa-comments-o"/> Contact Us</a></li></ul></nav></div></div></header><main class="container"><h1><i class="fa fa-bullhorn"/> Blog</h1><section><h2>By date of publication</h2><article typeof="BlogPosting"><h3 typeof="headline"><a typeof="url" href="user-stories-are-not-always-user-stories/">User Stories are not always User Stories</a></h3><div typeof="description"><p>Sometimes User Stories are not User Stories but something else: Epics, Tasks, Acceptance Criteria, or Definition of Done. By giving examples for these cases, I hope to help product owners, business analysts, requirements engineers, scrum masters and developers to improve their agile requirements engineering process. <span property="keywords" content="User Story,User Stories,Epic,Epics,Acceptance Criteria,Definition of Done,Definition of Ready,Requirements,Agile Requirements Engineering,DoR,DoD,INVEST"/></p></div><p><a href="user-stories-are-not-always-user-stories/">Read more…</a></p></article></section><section><h2>By name</h2><article typeof="BlogPosting"><h3 typeof="headline"><a typeof="url" href="user-stories-are-not-always-user-stories/">User Stories are not always User Stories</a></h3><div typeof="description"><p>Sometimes User Stories are not User Stories but something else: Epics, Tasks, Acceptance Criteria, or Definition of Done. By giving examples for these cases, I hope to help product owners, business analysts, requirements engineers, scrum masters and developers to improve their agile requirements engineering process. <span property="keywords" content="User Story,User Stories,Epic,Epics,Acceptance Criteria,Definition of Done,Definition of Ready,Requirements,Agile Requirements Engineering,DoR,DoD,INVEST"/></p></div><p><a href="user-stories-are-not-always-user-stories/">Read more…</a></p></article></section><section><h2>By author</h2><h3>Christian Hujer</h3><article typeof="BlogPosting"><h3 typeof="headline"><a typeof="url" href="user-stories-are-not-always-user-stories/">User Stories are not always User Stories</a></h3><div typeof="description"><p>Sometimes User Stories are not User Stories but something else: Epics, Tasks, Acceptance Criteria, or Definition of Done. By giving examples for these cases, I hope to help product owners, business analysts, requirements engineers, scrum masters and developers to improve their agile requirements engineering process. <span property="keywords" content="User Story,User Stories,Epic,Epics,Acceptance Criteria,Definition of Done,Definition of Ready,Requirements,Agile Requirements Engineering,DoR,DoD,INVEST"/></p></div><p><a href="user-stories-are-not-always-user-stories/">Read more…</a></p></article></section></main><footer class="row pagefooter"><div class="container clearfix"><div class="fLeft footerLeftWrap fLeftWrap"><span>© Copyright 2015-2016 Nelkinda Software Craft Private Limited. All rights reserved.</span></div><nav class="socialIcons fRight footerRightWrap fRightWrap"><a target="_blank" class="fo icon facebook" href="https://www.facebook.com/nelkinda"></a><a target="_blank" class="fo icon twitter" href="https://twitter.com/nelkinda"></a><a target="_blank" class="fo icon gplus" href="https://plus.google.com/102266732267155466362/about"></a><a target="_blank" class="fo icon email" href="mailto:info@nelkinda.com?Subject=Hello"></a><a target="_blank" class="fo icon github" href="https://github.com/orgs/nelkinda"></a><a target="_blank" class="fa icon bitbucket" href="https://bitbucket.org/nelkinda"></a></nav></div></footer><script src="/js/script"></script></body></html>
```

**Comment by christianhujer** [comment]
File RDFa_testcase.xhtml:
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
<head>
<title>Blog</title>
</head>
<body typeof="Blog" vocab="http://schema.org/">
<h1>Blog</h1>
<h2>By date of publication</h2>
<article typeof="BlogPosting">
<h3 typeof="headline"><a typeof="url" href="user-stories-are-not-always-user-stories/">User Stories are not always User Stories</a></h3>
<div typeof="description">
<p>
Sometimes User Stories […].
<span property="keywords" content="User Story,User Stories,Epic,Epics,Acceptance Criteria,Definition of Done,Definition of Ready,Requirements,Agile Requirements Engineering,DoR,DoD,INVEST"/>
</p>
</div>
</article>
</body>
</html>
```

**Comment by christianhujer** [comment]
I could reproduce the problem with:
* 16.12.18 (master commit id 2b6d521c912c2fb6e1bb7404eb44bb8b9836693f)
* 16.6.29 (latest release)
* 16.11.19 (latest dev release)

**Comment by Arttse** [comment]
Is `span` element have the `content` attribute? [Global attributes](https://html.spec.whatwg.org/multipage/dom.html#global-attributes)

I built the `vnu.jar` file from the latest for now 2b6d521 commit.

At the CLI I get the same errors:
```sh
$ java -jar /vnu.jar start.xhtml
"file:/[...]/start.xhtml":1.1838-1.2027: error: Attribute “content” not allowed on element “span” at this point.
"file:/[...]/start.xhtml":1.2858-1.2858: error: Attribute “content” not allowed on element “span” at this point.
"file:/[...]/start.xhtml":1.3526-1.3715: error: Attribute “content” not allowed on element “span” at this point.
"file:/[...]/start.xhtml":1.4179-1.4179: info warning: Document uses the Unicode Private Use Area(s), which should not be used in publicly exchanged documents. (Charmod C073)
```

If i run it as [Standalone web server](https://github.com/validator/validator#standalone-web-server):
```sh
$ java -cp vnu.jar nu.validator.servlet.Main 8888
```

and upload file `start.xhtml` to validator:
![image](https://cloud.githubusercontent.com/assets/7123950/21292803/b3f726c8-c521-11e6-9bed-3aa5f50099eb.png)

**Comment by christianhujer** [comment]
Note that `content` is not an _HTML_ attribute but an _RDFa_ attribute.

**Comment by sideshowbarker** [comment]
The `content` attribute is not part of RDFa Lite but instead part of full RDFa.

The release versions of the `vnu.jar` (offline) checker only support RDFa Lite and intentionally do not support full RDFa. Neither does the https://checker.html5.org/ online checker. The only instance of the checker that does support full RDFa is the backend of https://validator.w3.org/nu/ (which is the same backend used if you check at  document with the legacy frontend at https://validator.w3.org/).

Getting a `vnu.jar` with full RDFa support requires building it from the sources like this:

```
python build/build.py --follow-w3c-spec update dldeps build jar
```

That is, the `--follow-w3c-spec` option must be specified.

After running that command, the result can be found at `./build/dist/vnu.jar`.

**Comment by sideshowbarker** [comment]
I don’t have any more info to add on this from my side, so I’m going ahead and closing this. But feel free to re-open it if needed.

**Comment by bazzadp** [comment]
I can't get this to work. Exact same scenario as above.

I have content attribute on a span tag:
    "file:/c:/temp/index.html":163.21-163.74: error: Attribute “content” not allowed on element “span” at this point.

I build using the `--follow-w3c-spec` tag and rerun against this new jar, but it still shows this same error. There were no errors in build process. Even tried from a completely fresh machine (Windows 7, Java 1.8.0_102, Validator 17.2.1) and same issue.

Error shows on the https://checker.html5.org/ online checker but not the https://validator.w3.org/nu/ checker.

Any ideas?

**Comment by sideshowbarker** [comment]
I think there’s one other thing that’s necessary which I neglected to mention: Run the jar like this:

```bash
java -jar --schema http://s.validator.nu/html5-all.rnc FILE.html
```

That is, specify the `--schema` option with the argument `http://s.validator.nu/html5-all.rnc`.

**Comment by bazzadp** [comment]
Yes that works now. Thanks.

But does give me some other (real!) errors that for some reason weren't showing before. Guessing it's a but more strict now. Will work through them...
